### DISCUSSION TITLE
questions about the extensions organization

### DISCUSSION THREAD

**Discussion by VictorB0** [vkhorikov/CSharpFunctionalExtensions#550] [General]
Good day, i have been taking a look on the source code and i have noticed that some async extensions like the 'Compensate' are in the 'ResultExtensions' while some others are in their respective 'AsyncResultExtensions(left, right or both)Operand' is there any reason for that, or is it  just part of the legacy of the project?
I have also noticed that some are in the unintended class, like the 'Match.Task.Right' is in the 'AsyncResultExtensionsLeftOperand' class, this probably would not make any difference in the use of the extensions, unless the user prefer to call the class explicitly.
I have never used value tasks, but i have noticed that async 'OnSuccessTry' does not have their respective value tasks counterpart, same for the 'TapIfTry' and 'TapTry'.
I have been using these extensions for a while now, glad the project exists, cheers.

**Comment by vkhorikov** [comment]
No, these inconsistencies are due to an oversight. There were a couple of waves of refactoring to organize these extension methods and it looks like not all the method were designated to the new structure properly. Also, `OnSuccessTry` should be renamed into `MapTry` or `BindTry` depending on the signature. All of the `OnSuccess` methods are deprecated.

**Reply by VictorB0** [reply]
Makes sense, i have never contributed to any open source project, and i am not entirely sure on how to report these details, or if they are even useful in the first place.
It seems that `OnFailure` also needs to be moved to the obsolete folder, i am still getting used to the new counterparts.
By the way, is there any reason why the `TapTry` and `TapIfTry` left async operand  does not have a 'errorHandler' parameter?
And again, thanks for maintaining this project, hope you are doing well.

**Reply by vkhorikov** [reply]
`OnFailure` -- yep, that one should be deprecated too

No particular reason, most likely an oversight as well.
